### PR TITLE
[IniSerializer] Add `serialize` to ini

### PR DESCRIPTION
**Pull request by zhoonit**
- [IniSerializer] Add `serialize` to ini 

```
**Changes proposed in this PR:**
- implements ini interpreter::serialize
- add corresponding test (reference shoud be equal to
serialize->deserialized graph)
- Fix some const correctness

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: nntrainer/compiler/ini_interpreter.cpp [hunk 1]
```diff
@@ -275,12 +276,21 @@ void IniGraphInterpreter::serialize(
...
+    layer->export_to(e);
+
+    const auto key_val_pairs =
+      e.get_result<ExportMethods::METHOD_STRINGVECTOR>();
+
+    for (const auto &pair : key_val_pairs) {
+      s.setEntry(pair.first, pair.second);
+    }
+
+    sections.push_back(s);
```
Comment (jijoongmoon) [inline nntrainer/compiler/ini_interpreter.cpp:289]: Just for a reminder, the order of sections is matters.

File: test/unittest/compiler/unittest_interpreter.cpp [hunk 2]
```diff
@@ -45,6 +45,41 @@ const std::string pathResolver(const std::string &path) {
...
+    nntrainer::Exporter rhs_export;
+
+    l.export_to(lhs_export);
+    r.export_to(rhs_export);
+
+    /*** fixme, there is one caveat that order matters in this form */
+    EXPECT_EQ(
+      lhs_export.get_result<nntrainer::ExportMethods::METHOD_STRINGVECTOR>(),
+      rhs_export.get_result<nntrainer::ExportMethods::METHOD_STRINGVECTOR>());
+  };
```
Comment (jijoongmoon) [inline test/unittest/compiler/unittest_interpreter.cpp:74]: I wonder how we define "equal two network graphs"?

File: nntrainer/compiler/ini_interpreter.cpp (hunk 1 above)
Comment (zhoonit) [reply nntrainer/compiler/ini_interpreter.cpp:289]: I think activation should exist as a layer when we serialize (we have an assumption that it's been already trained). 
And I agree that time disributed layer should be considered later.

File: test/unittest/compiler/unittest_interpreter.cpp (hunk 2 above)
Comment (zhoonit) [reply test/unittest/compiler/unittest_interpreter.cpp:74]: I agree, making a graph equal have some different way to define.
For now, I don't have any idea to cover all the requirements, so didn't actually make an equal operator.

My thinking is that we could make a discussion if equal operator is needed in the future.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by lhs8928 — APPROVED** [review]
LGTM

**Pull request merged**